packages/common/aws.js

```diff
@@ -972,16 +974,34 @@ exports.isThrottlingException = (err) => err.code === 'ThrottlingException';
  * Given a Cumulus step function event, if the message is on S3, pull the full message
  * from S3 and return, otherwise return the event.
  *
- * @param {Object} event - the Cumulus event
+ * @param {Object} incomingEvent - the Cumulus event
  * @returns {Object} - the full Cumulus message
  */
-exports.pullStepFunctionEvent = async (event) => {
-  if (event.replace) {
-    const file = await exports.getS3Object(event.replace.Bucket, event.replace.Key);
-
-    return JSON.parse(file.Body.toString());
+exports.pullStepFunctionEvent = async (incomingEvent) => {
+  if (!incomingEvent.replace) {
+    return incomingEvent;
+  }
+  const event = cloneDeep(incomingEvent);
+  const remoteMsgS3Object = await exports.getS3Object(event.replace.Bucket, event.replace.Key);
+  const remoteMsg = JSON.parse(remoteMsgS3Object.Body.toString());
+
+  let returnEvent = remoteMsg;
+  if (event.replace.TargetPath) {
```

Comment:
Should we have concerns here with backward compatibility at all? I'm leaning towards 'no', but wanted to ask.

Reply:
Hmm, isn't this backward compatible?       If there's no targetPath it returns the value of JSON.parse(remoteMsgS3Object.Body.toString()); like before.

Reply:
Ah, yeah, misread it. Good! 👍